angel_system/eval/compute_scores.py

```diff
@@ -8,21 +8,23 @@ def iou_per_activity_label(labels, gt, dets):
     """
     Calculate the iou per activity label
 
-    :param labels: List of String labels of the activity classes
+    :param labels: Pandas df with columns id (int) and class (str)
     :param gt: Dict of activity start and end time ground truth values, organized by label keys
     :param dets: Dict of activity start and end time detections with confidence values, organized by label keys
 
-    :return: Tuple(Average IoU across all classes, Dictionary mapping class labels to their average IoU scores)
+    :return: Tuple(Average IoU across all classes, Dictionary mapping class labels to their average IoU scores, dets)
```

Comment:
I was returning the dets back out so the visualization code wouldn't have to repeat the "detection intersection" calculations. I could just call the detection intersection calculations again from the visualization code (which would help ensure that dets actually has the "detection intersection" set instead of assuming that this was already called)? I will add the note about modifying in-place though

Reply:
You are correct in that the visualization code should not repeat the detection intersection again. I would say remove the `dets` return from here. This function modifies the input data structure, which is likely sufficient. Just add comments enough to describe what this function is doing to the `dets` structure.